File: system-test/java/src/main/java/com/mycompany/myshop/testkit/dsl/core/usecase/usecases/GoToMyShop.java
package com.mycompany.myshop.testkit.dsl.core.usecase.usecases;

import com.mycompany.myshop.testkit.driver.port.MyShopDriver;
import com.mycompany.myshop.testkit.driver.port.dtos.GoToMyShopRequest;
import com.mycompany.myshop.testkit.dsl.core.usecase.usecases.base.BaseMyShopUseCase;
import com.mycompany.myshop.testkit.dsl.core.shared.UseCaseResult;
import com.mycompany.myshop.testkit.dsl.core.shared.UseCaseContext;
import com.mycompany.myshop.testkit.dsl.core.shared.VoidVerification;

public class GoToMyShop extends BaseMyShopUseCase<Void, VoidVerification> {
    public GoToMyShop(MyShopDriver driver, UseCaseContext context) {
        super(driver, context);
    }

    @Override
    public UseCaseResult<Void, VoidVerification> execute() {
        var request = GoToMyShopRequest.builder().build();
        var result = driver.goToMyShop(request).mapVoid();
        return new UseCaseResult<>(result, context, VoidVerification::new);
    }
}

File: system-test/typescript/src/testkit/dsl/core/usecase/usecases/GoToMyShop.ts
import type { MyShopDriver } from '../../../../driver/port/my-shop-driver.js';
import { UseCaseResult } from '../../shared/use-case-result.js';
import { VoidVerification } from '../../shared/void-verification.js';
import type { UseCaseContext } from '../../shared/use-case-context.js';
import { BaseMyShopUseCase } from './base/BaseMyShopUseCase.js';

export class GoToMyShop extends BaseMyShopUseCase<void, VoidVerification> {
  constructor(driver: MyShopDriver, context: UseCaseContext) {
    super(driver, context);
  }

  async execute(): Promise<UseCaseResult<void, VoidVerification>> {
    const result = await this.driver.goToMyShop();

    return new UseCaseResult(
      result,
      this.context,
      (_, ctx) => new VoidVerification(undefined, ctx),
    );
  }
}
